### ISSUE TITLE
Program crashes during attempt to play certain stream links.

### ISSUE THREAD

**Issue by BoltSwith** [schreibfaul1/ESP32-audioI2S#930]
Greetings,

I have one issue and one questions.

ISSUE:

I upgraded to Lib version:  Version 3.0.13x few days ago. When attempting to stream a radio station using connectToHost(), the programme crashes upon attempting to connect to the host. 

This happens particularly when I connect to - ConnectToHost("http://stream.antennethueringen.de/live/aac-64/stream.antennethueringen.de/");

Other radio stations such as the following plays fine:
- audio.connecttohost("http://www.wdr.de/wdrlive/media/einslive.m3u"); 
 

I get the following crash report on the console when the program crashes:

```
Connected to host: TRUE 
#############Free SRAM: 38647 bytes
Guru Meditation Error: Core  0 panic'ed (StoreProhibited). Exception was unhandled.

Core  0 register dump:
PC      : 0x42160f6a  PS      : 0x00060030  A0      : 0x82037f80  A1      : 0x3fca8c90
--- 0x42160f6a: ComplexMult(long*, long*, long, long, long, long) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.h:380

A2      : 0x00000000  A3      : 0x00000000  A4      : 0x00000000  A5      : 0x00000000
A6      : 0x7fffff62  A7      : 0x000c90fe  A8      : 0x00000004  A9      : 0x00000000
A10     : 0x3c23fcb4  A11     : 0x00002000  A12     : 0x80384545  A13     : 0x3fca3000
A14     : 0x00060023  A15     : 0x00002000  SAR     : 0x00000003  EXCCAUSE: 0x0000001d
EXCVADDR: 0x00000004  LBEG    : 0x400570e8  LEND    : 0x400570f3  LCOUNT  : 0x00000000
--- 0x400570e8: memset in ROM
0x400570f3: memset in ROM



Backtrace: 0x42160f67:0x3fca8c90 0x42037f7d:0x3fca8cb0 0x42038406:0x3fca8cf0 0x420384de:0x3fca8d10 0x4203bd32:0x3fca8d90 0x4203bee2:0x3fca8de0 0x4203bfe6:0x3fca8e10 0x4203c129:0x3fca8e40 0x4203c332:0x3fca8e60 0x4203c4dd:0x3fca8ed0 0x42034d9e:0x3fca8ef0 0x4202c115:0x3fca8f30 0x4202c7cd:0x3fca8fa0 0x4202c969:0x3fca8fc0
--- 0x42160f67: ComplexMult(long*, long*, long, long, long, long) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.h:380
0x42037f7d: faad_imdct(mdct_info*, long*, long*) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:3906 (discriminator 3)
0x42038406: imdct_long(fb_info*, long*, long*, unsigned short) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:4076
0x420384de: ifilter_bank(fb_info*, unsigned char, unsigned char, unsigned char, long*, long*, long*, unsigned char, unsigned short) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:4141
0x4203bd32: reconstruct_channel_pair(NeAACDecStruct*, ic_stream*, ic_stream*, element*, short*, short*) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:7480
0x4203bee2: channel_pair_element(NeAACDecStruct*, _bitfile*, unsigned char, unsigned char*) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:8169
0x4203bfe6: decode_cpe(NeAACDecStruct*, NeAACDecFrameInfo*, _bitfile*, unsigned char) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:7881
0x4203c129: raw_data_block(NeAACDecStruct*, NeAACDecFrameInfo*, _bitfile*, program_config*, drc_info*) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:7916
0x4203c332: aac_frame_decode(NeAACDecStruct*, NeAACDecFrameInfo*, unsigned char*, unsigned long, void**, unsigned long) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:1099
0x4203c4dd: NeAACDecDecode2(void*, NeAACDecFrameInfo*, unsigned char*, unsigned long, void**, unsigned long) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/libfaad/neaacdec.cpp:990
0x42034d9e: AACDecode(unsigned char*, long*, short*) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/aac_decoder/aac_decoder.cpp:151
0x4202c115: Audio::sendBytes(unsigned char*, unsigned int) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/Audio.cpp:4439
0x4202c7cd: Audio::playAudioData() at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/Audio.cpp:3642 (discriminator 2)
 (inlined by) Audio::playAudioData() at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/Audio.cpp:3614 (discriminator 2)
0x4202c969: Audio::performAudioTask() at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/Audio.cpp:6350
 (inlined by) Audio::performAudioTask() at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/Audio.cpp:6343
 (inlined by) Audio::audioTask() at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/Audio.cpp:6338
 (inlined by) Audio::taskWrapper(void*) at D:/ESP5_PROJS/PIXLPAL-M1/components/ESP32-audioI2S/src/Audio.cpp:6332





ELF file SHA256: 78e918b84

Rebooting...
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x3b (SPI_FAST_FLASH_BOOT)
Saved PC:0x40375dd5
--- 0x40375dd5: esp_restart_noos at D:/ESP_IDF_53/v5.3.1/esp-idf/components/esp_system/port/soc/esp32s3/system_internal.c:158
```

**Comment by BoltSwith** [comment]
My 2nd Question is, Can I reduce the 32 number of i2s DMA buffer without sacrificing audio output qualitiy?

My program is short of SRAM. What is the implication of reducing the i2s DMA Buffer from 32 to say 10 - 16?

**Comment by BoltSwith** [comment]
I've resolved the issue of Program Crash when playing the station - http://stream.antennethueringen.de/live/aac-64/stream.antennethueringen.de/

It appears the issue arose from the Arduino Lib Component I was using with the esp-idf i.e. arduino-esp32 version: v3.1.0-rc1

The component prevented other arduino libraries from detecting/accessing the 8mb PSRAM of the ESP32S3 SoC.

I upgraded the arduino-esp32 Component (v3.1.0~1) which was released 10hours ago, and re-ran the programs that use the AudioI2S Lib. Know issues have been resolved as a result of the upgrade. PSRAM is being detected/accessed.


As for the issue with i2s DMA Buffer, it's been resolved too. I was initially using the ESP32 Bluedroid Library, I switched to the memory efficient NimBLE Library. This solved the issue.
